### ISSUE TITLE
关于数据增强后seq_len的问题

### ISSUE THREAD

**Issue by hua-ops** [ShusenTang/WriteLikeYou#4]
您好，我最近正在做笔迹生成相关方面的工作，有幸拜读了您的论文，受益匪浅！
但我发现在`./data.py`的`_processing_zi(self, zi)`函数中，如果进行了`augment_strokes`的数据增强，那么最终的数据序列长度可能会变短，但`seq_len`仍然使用的是数据增强之前的长度，而当我将`seq_len`修改为数据增强之后的序列长度后，模型效果却出现较大的下降，请问你们当时有出现过这种问题吗？

**Comment by mio-tsuki** [comment]
**个人理解如下**：

- 如果按照原`seq_len`计算的话，实际`Ls numerator`按`augment_strokes`后较短序列算，`denominator`按`augment_strokes`前较长长度除，相当于给经过`augment_strokes`处理后的数据降权了，如果不降权的话会导致模型学到很多短序列的脏数据，让模型拟合这些不太对的数据导致生成质量下降

```python
self.Ls = tf.reduce_sum(Ls) / tf.to_float(tf.reduce_sum(self.dec_seq_lens))
```

- 同时，`seq_len`也传给`content`/`style encoder`的`bidirectional_dynamic_rnn`和`attention mask`。`content`/`style encoder` 是预训练好的，主要见的是正常长度分布。`augment_strokes`处理后长度变短后输入，会让 encoder 的 `final state`、`attention memory` mask都不对，这对`frozen encoder`是分布漂移。保持原有长度会让RNN多弄到一些 zero padding，弱化这个分布漂移

- 而且Lp训练的时候没有用`seq_len` mask，改了之后Ls和Lp的就不一样了

---

~~另外我实测的话，先无`augment_strokes`训练到大概能看再加`augment_strokes`继续训效果比一开始就开`augment_strokes`更好~~

~~嗯就是个人理解吧，如果有错误不要喷啊啊啊，我就一个苦逼高中牲~~
